apps/drec-ui/src/apps/device-group/view/pages/ReservedPage/ReservedPage.effects.ts

```diff
@@ -0,0 +1,99 @@
+import { DeviceGroupModalsActionsEnum, useDeviceGroupModalsDispatch } from '../../context';
+import {
+    UnreservedFormFormValues,
+    useSelectableDeviceGroupsTableLogic,
+    useDeviceGroupsFilterFormLogic
+} from '../../../logic';
+import { GenericFormProps } from '@energyweb/origin-ui-core';
+import { useAllDeviceFuelTypes, useReservedDeviceGroups } from '../../../data';
+import { useEffect, useState } from 'react';
+import { SelectableDeviceGroupDTO } from '@energyweb/origin-drec-api-client';
+
+export const useReservedPageEffects = () => {
+    const initialFormValues: UnreservedFormFormValues = {
+        country: [],
+        fuelCode: [],
+        installationConfiguration: '',
+        offTaker: '',
+        sector: '',
+        standardCompliance: '',
+        gridInterconnection: '',
+        commissioningDateRange: '',
+        capacityRange: ''
+    };
+    const [filterUnreserved, setFilterUnreserved] = useState(initialFormValues);
+    const { deviceGroups, isLoading: IsLoadingDeviceGroups } =
+        useReservedDeviceGroups(filterUnreserved);
+    const [selectedDeviceGroupList, setSelectedDeviceGroupList] = useState(deviceGroups);
+    const dispatchModals = useDeviceGroupModalsDispatch();
+
+    const { allTypes, isLoading: isDeviceTypesLoading } = useAllDeviceFuelTypes();
+
+    const { fields, initialValues, validationSchema, buttonText } = useDeviceGroupsFilterFormLogic(
+        filterUnreserved,
```

Comment:
`initialFormValues` should be passed here

Reply:
I am passing the filterUnreserved which initially contains the initialFormValues. Because on each state change, I ned the form to be updated.

Reply:
`initialValues` of the form are set only once on the first render. Passing it on the later stages won't affect the already set values. And it is updating because the form handles the state on its own after initial render